### ISSUE TITLE
machine.mem32[] returns signed 32bit integers

### ISSUE THREAD

**Issue by oddstr13** [micropython/micropython-esp32#81]
I would expect unsigned integers here.

**Comment by dpgeorge** [comment]
Yes, it's possible that unsigned integers would be the more common use.  But it's easy to get around the current behaviour: just do `mem32[addr] & 0xffffffff`.

It could be argued that there should be signed and unsigned accessors, like mem32 for signed and umem32 for unsigned.

**Comment by nickzoic** [comment]
(yeah, I think we should fix that ... but that'd be all the ports, right?
